Zend/zend_portability.h

```diff
@@ -350,13 +350,25 @@ char *alloca();
 # define ALLOCA_FLAG(name) \
 	bool name;
 # define SET_ALLOCA_FLAG(name) \
-	name = 1
+	name = true
 # define do_alloca_ex(size, limit, use_heap) \
 	((use_heap = (UNEXPECTED((size) > (limit)))) ? emalloc(size) : alloca(size))
 # define do_alloca(size, use_heap) \
 	do_alloca_ex(size, ZEND_ALLOCA_MAX_SIZE, use_heap)
 # define free_alloca(p, use_heap) \
 	do { if (UNEXPECTED(use_heap)) efree(p); } while (0)
+#elif defined(PHP_WIN32)
```

Comment:
I don't think special casing for Windows is unnecessary. Using `_malloca()` instead of `_alloca()` would defeat the purpose of our elaborate stack vs. heap allocation, since `_malloca()` already does this, and apparently the threshold is 1024 (much less than the 32*1024 we're using). So sticking with `_alloca()` appears to be good; and since we don't forbid using the standard C names, `alloca` is defined as alias of `_alloca()`. In other words, the Windows side doesn't require special treatment here; we didn't use main/alloca.c there.

Reply:
I'll drop those specific Windows changes then, looking at it I thought it didn't work.

Reply:
I've double-checked with a standalone program:
````c
#include <stdio.h>
#include <malloc.h>

int main()
{
    void *p = alloca(100);
    return p == NULL;
}
````
````
>alloca.exe
>echo %errorlevel%
0
````
and also had a look at the SDK headers.